src/components/widgets/stackedColumn/stackedColumn_dataHelpers.js

```diff
@@ -0,0 +1,137 @@
+
+// todo - not import Highcharts again
+import Highcharts from 'highcharts';
+import last from 'lodash/last';
+
+
+export const makeChartOptions = ({
+  emitSetState = () => {},
+  widget,
+}) => {
+
+  const categories = Highcharts.getOptions().lang.shortMonths;
+
+  return {
+    // default column options
+    chart: {
+      type: 'column',
+      events: {
+        load: function() {  // equivalent to constructor callback
+          var seriesData = this.series[0].data;//this is series data
+          seriesData.forEach((d, idx) => {
+            if (d.y === null) { //find null value in series
+              // adds plot band
+              this.xAxis[0].addPlotBand({
+                from: idx -.5,  // point back
+                to: idx + .5,   // point after
+                color: 'url(#diagonal-stripe-1)', // this color represents the null value region
+              });
+            }
+          });
+
+          let customLegendData = this.series.map(s => {
+            const lastData = last(s.data);
+            return {
+              // key: lastData.category,
+              key: s.name,
+              y: lastData.y,
+              seriesName: s.name,
+              color: lastData.color,
+            }
+          });
+          emitSetState({'customLegend': customLegendData});
+
+          // "hover" over the last column
+          const lastCol = last(this.series[0].data);
+          if (lastCol) {
+            lastCol.onMouseOver && lastCol.onMouseOver();
+          }
+        },
+      },
+    },
+    title: {
+      text: widget.title,
+      align: 'left',
+    },
+    subtitle: {
+      useHTML: true,
+      text: `<span>Last updated <time dateTime="${widget.dateUpdated}">${widget.dateUpdated}</time></span>`,
+      align: 'left',
+    },
+    plotOptions: {
+      column: {
+        stacking: widget.percentageStacking ? 'percent' : 'normal'
+      },
+      series: {
+        animation: false,
+        point: {
+          events: {
+            mouseOver: function() {
+              const sliceIdx = this.index;
+              const chartSeries = this.series.chart.series
+              const customLegendData = this.series.chart.series.map((s, i) => {
```

Comment:
this smells funny, we can dry up replace `this.series.chart.series` with `chartSeries` and `key: s.name`